src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (c >= a + b || b >= a + c || a >= b + c) {
+      throw new Error('Error');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Error');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const aDiff = s - this.a;
+    const bDiff = s - this.b;
+    const cDiff = s - this.c;
 
+    return Math.round(Math.sqrt(s * aDiff * bDiff * cDiff) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Error. The radius is less than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Error. One of sids is less than 0');
+    }
+  }
+
+  getArea(): number {
+    return this.width * this.height;
```

Comment:
what if the width or height is in the following format: _41.345_ and has thousands?
method `getArea` have to return a number rounded down to hundredths. 
```suggestion
    return Math.floor(this.width * this.height) / 100;
```